tests/periph_timer_cli/main.c

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for periph timer API
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+
+#include "shell.h"
+#include "periph/timer.h"
+#include "periph/gpio.h"
+#include "mutex.h"
+
+#define ARG_ERROR       (-1)
+#define CONVERT_ERROR   (-32768)
+#define RESULT_OK       (0)
+#define RESULT_ERROR    (-1)
+#define INVALID_ARGS    puts("Error: Invalid number of arguments");
+
+#define CB_TOGGLE_STR   "cb_toggle"
+#define CB_HIGH_STR     "cb_high"
+#define CB_LOW_STR      "cb_low"
+
+static mutex_t cb_mutex;
+
+static inline int _get_num(const char *str)
+{
+    errno = 0;
+    char *temp;
+    long val = strtol(str, &temp, 0);
+
+    if (temp == str || *temp != '\0' ||
+        ((val == LONG_MIN || val == LONG_MAX) && errno == ERANGE)) {
+        val = CONVERT_ERROR;
```

Comment:
the error case is not evaluated in many cases where this function is used. Maybe change to
```
static inline int _get_num(const char *str, uint32_t *val)
```

and return value by reference and error as return?

Reply:
done